server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2730,6 +2730,37 @@ public void updateTabletDataFile(long maxCommittedTime, FileRef newDatafile, Fil
 
   }
 
+  public void updateLastLocation(){
+    synchronized (timeLock) {
```

Comment:
why synchronize on timeLock.  That is used for controlling access to persistedTime.

Reply:
That was leftover past iterations of my testing. I suppose if I update perssitedTime inside updateLastLocation as you mentioned below it might be needed?